src/Uno.UI/UI/Xaml/Media/ImageSource.iOS.cs

```diff
@@ -190,44 +202,65 @@ internal async Task<UIImage> Open(CancellationToken ct)
 		/// </summary>
 		private async Task OpenUsingPlatformDownloader(CancellationToken ct)
 		{
-			if (SupportsAsyncFromBundle)
-			{
-				DownloadUsingPlatformDownloader();
-			}
-			else
+			if (ct.IsCancellationRequested)
 			{
-				await CoreDispatcher.Main.RunAsync(
-					CoreDispatcherPriority.Normal,
-					() =>
-					{
-						DownloadUsingPlatformDownloader();
-					}
-				).AsTask(ct);
+				return;
 			}
-		}
 
-		private void DownloadUsingPlatformDownloader()
-		{
 			using (var url = new NSUrl(WebUri.OriginalString))
 			{
-				NSError error;
-
-				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				using (var request = NSUrlRequest.FromUrl(url))
 				{
-					this.Log().Debug($"Loading image from [{WebUri.OriginalString}]");
-				}
+					NSUrlSessionDataTask task;
+					var awaitable = DefaultSession.CreateDataTaskAsync(request, out task);
+					ct.Register(OnCancel);
+					try
+					{
+						task.Resume(); // We need to call this manually https://bugzilla.xamarin.com/show_bug.cgi?id=28425#c3
+						var result = await awaitable;
+						task = null;
+						var response = result.Response as NSHttpUrlResponse;
+
+						if (ct.IsCancellationRequested)
+						{
+							return;
+						}
+						else if (!IsSuccessful(response.StatusCode))
+						{
+							if (this.Log().IsEnabled(LogLevel.Error))
+							{
+								this.Log().LogError(NSHttpUrlResponse.LocalizedStringForStatusCode(response.StatusCode));
+							}
 
-				// fallback on the platform's loader
-				using (var data = NSData.FromUrl(url, NSDataReadingOptions.Coordinated, out error))
-				{
-					if (error != null)
+						}
+						else
+						{
+							ImageData = UIImage.LoadFromData(result.Data);
+						}
+					}
+					catch (NSErrorException e)
 					{
-						this.Log().Error(error.LocalizedDescription);
+						// This can occur for various reasons: download was cancelled, NSAppTransportSecurity blocks download, host couldn't be resolved...
+						if (ct.IsCancellationRequested)
+						{
+							if (this.Log().IsEnabled(LogLevel.Debug))
+							{
+								this.Log().LogDebug(e.ToString());
+							}
+						}
+						else if (this.Log().IsEnabled(LogLevel.Error))
+						{
+							this.Log().LogError(e.ToString());
+						}
 					}
-					else
+
+					void OnCancel()
 					{
-						ImageData = UIImage.LoadFromData(data);
+						// Cancel the current download
+						task?.Cancel();
 					}
+
+					bool IsSuccessful(nint status) => status < 300 || status == 304;
```

Comment:
That would be a strange behavior if your get a `304` here, because it means you have no payload to display.

Reply:
Fixed